### PR TITLE
feat(resetPass): add reset password interface

### PR DESCRIPTION
**Pull request by AzizSandid**
## What does this do?

create  reset password interface 

## Why did you do this?

To allow users to reset their passwords easily and securely.

## How did you test this?
Screenshot_20240704-122939_archimatch
Screenshot_20240704-122738_archimatch
Screenshot_20240704-122724_archimatch
Screenshot_20240704-122713_archimatch

### CODE REVIEW

File: src/assets/icons/archimatch/design-inter.tsx [hunk 1]
```diff
@@ -2,14 +2,8 @@ import * as React from 'react';
...
-const DesignInter  = (props: SvgProps) => (
-  <Svg
-    
-    width={115}
-    height={104}
-    fill="none"
-    {...props}
-  >
+const DesignInter = (props: SvgProps) => (
+  <Svg width={115} height={104} fill="none" {...props}>
```
Comment (MohamedMegdich1) [inline src/assets/icons/archimatch/design-inter.tsx:6]: Please remember not to set fixed width and height.

File: src/assets/icons/archimatch/house-model.tsx [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
 
-const HouseModel  = (props: SvgProps)  => (
-  <Svg 
-    width={100}
-    height={100}
-    fill="none"
-    {...props}
-  >
+const HouseModel = (props: SvgProps) => (
+  <Svg width={100} height={100} fill="none" {...props}>
```
Comment (MohamedMegdich1) [inline src/assets/icons/archimatch/house-model.tsx:6]: the same

File: src/assets/icons/tick-icon.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import * as React from 'react';
+import type { SvgProps } from 'react-native-svg';
+import Svg, { Path } from 'react-native-svg';
+
+interface TickIconProps extends SvgProps {
+  color?: string;
+}
+
+const TickIcon: React.FC<TickIconProps> = ({ color = '#000', ...props }) => (
+  <Svg width={24} height={24} fill="none" viewBox="0 0 24 24" {...props}>
```
Comment (MohamedMegdich1) [inline src/assets/icons/tick-icon.tsx:11]: the same

File: src/assets/images/CodeCerificationBlue.png
Comment (MohamedMegdich1) [inline src/assets/images/CodeCerificationBlue.png:1]: Watch out for the file naming.

File: src/assets/images/CodeCerificationRed.png
Comment (MohamedMegdich1) [inline src/assets/images/CodeCerificationRed.png:1]: the same kebab case

File: src/modules/reset-password/otp-input.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import type { Control, FieldValues, Path } from 'react-hook-form';
+import { useController } from 'react-hook-form';
+import { TextInput, View } from 'react-native';
+
+type OTPInputProps<T extends FieldValues> = {
+  length?: number;
+  disabled: boolean;
+  name: Path<T>;
+  control: Control<T>;
+  rules?: any;
```
Comment (MohamedMegdich1) [inline src/modules/reset-password/otp-input.tsx:11]: why any type ?

File: src/modules/reset-password/reset-form-email.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import { useForm } from 'react-hook-form';
+import * as z from 'zod';
+
+import { Button, ControlledInput, View } from '@/shared/components';
+import { translate } from '@/translations/i18n';
+
+const schema = z.object({
+  email: z
+    .string({
+      required_error: 'Email is required',
```
Comment (MohamedMegdich1) [inline src/modules/reset-password/reset-form-email.tsx:13]: Please translate the message and make sure to use the validation that has been shared.

File: src/modules/reset-password/reset-form-password.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        <PasswordRequirementItem
+          isValid={/[A-Z]/.test(password)}
+          message="resetpass.password-uppercase"
+        />
+        <PasswordRequirementItem
+          isValid={/[0-9]/.test(password)}
+          message="resetpass.password-digit"
+        />
+        <PasswordRequirementItem
+          isValid={/[^a-zA-Z0-9]/.test(password)}
```
Comment (MohamedMegdich1) [inline src/modules/reset-password/reset-form-password.tsx:55]: Do you think this is the best method to do it? I think you can define a list and use it instead.

File: src/modules/reset-password/reset-stepper.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        className="flex-1 p-6 pt-12"
+        contentContainerClassName={`${step === 1 && 'items-center'}`}
+      >
+        {step === 1 && (
+          <Image
+            className="mb-6 h-24 w-44 self-center overflow-hidden"
+            contentFit="cover"
+            source={
+              space === 'client'
+                ? require('@/assets/images/CodeCerificationBlue.png')
```
Comment (MohamedMegdich1) [inline src/modules/reset-password/reset-stepper.tsx:68]: Why not export the SVG image and change the color based on the context?

File: src/modules/reset-password/verification-code.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import type { SubmitHandler } from 'react-hook-form';
+import { useForm } from 'react-hook-form';
+import * as z from 'zod';
+
+import { Button, Text, TouchableOpacity, View } from '@/shared/components';
+import { translate } from '@/translations/i18n';
+
+import { OTPInput } from './otp-input';
+
+const schema = z.object({
```
Comment (MohamedMegdich1) [inline src/modules/reset-password/verification-code.tsx:12]: the same for the schema

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/assets/icons/archimatch/design-inter.tsx (hunk 1 above)
Comment (AzizSandid) [reply src/assets/icons/archimatch/design-inter.tsx:6]: fixed

File: src/assets/icons/archimatch/house-model.tsx (hunk 2 above)
Comment (AzizSandid) [reply src/assets/icons/archimatch/house-model.tsx:6]: fixed

File: src/assets/icons/tick-icon.tsx (hunk 3 above)
Comment (AzizSandid) [reply src/assets/icons/tick-icon.tsx:11]: fixed

File: src/assets/images/CodeCerificationBlue.png
Comment (AzizSandid) [reply src/assets/images/CodeCerificationBlue.png:1]: fixed

File: src/assets/images/CodeCerificationRed.png
Comment (AzizSandid) [reply src/assets/images/CodeCerificationRed.png:1]: fixed

File: src/modules/reset-password/otp-input.tsx (hunk 4 above)
Comment (AzizSandid) [reply src/modules/reset-password/otp-input.tsx:11]: fixed

File: src/modules/reset-password/reset-form-email.tsx (hunk 5 above)
Comment (AzizSandid) [reply src/modules/reset-password/reset-form-email.tsx:13]: fixed

File: src/modules/reset-password/verification-code.tsx (hunk 8 above)
Comment (AzizSandid) [reply src/modules/reset-password/verification-code.tsx:12]: fixed

File: src/modules/reset-password/reset-stepper.tsx (hunk 7 above)
Comment (AzizSandid) [reply src/modules/reset-password/reset-stepper.tsx:68]: image
it's image nt svg i can't change the color

Comment (guardian-001) [reply src/modules/reset-password/reset-stepper.tsx]: Basically, it's a PNG image, but when exported from Figma as an SVG, it just encapsulates the PNG image in an SVG file

File: src/assets/icons/tick-icon.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+// TickIcon.tsx
+import * as React from 'react';
+import type { SvgProps } from 'react-native-svg';
+import Svg, { Path } from 'react-native-svg';
+
+interface TickIconProps extends SvgProps {
+  color?: string;
+}
+
+const TickIcon: React.FC<TickIconProps> = ({ color = '#000', ...props }) => (
```
Comment (MohamedMegdich1) [inline src/assets/icons/tick-icon.tsx:10]: Change the color to a non-static color.

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/assets/icons/tick-icon.tsx (hunk 9 above)
Comment (AzizSandid) [reply src/assets/icons/tick-icon.tsx]: fixed

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**